### DISCUSSION TITLE
using async function in app.onError

### DISCUSSION THREAD

**Discussion by AliKaanT** [honojs/hono#991] [Q&A] [answered]
Hi. I need to await my function inside of app.onError(). But i cant make my function async so it doesnt work. Can we use async function in it or any other ideas ?
![image](https://user-images.githubusercontent.com/90530758/226041737-5c78131f-3b80-41d1-9c54-3f060baa2a1d.png)

**Comment by whilenot-dev** [comment]
one idea i would have is to use a middleware and just rethrow the error:
```typescript
app.use("*", async (c, next) => {
  try {
    await next();
  } catch (err) {
    await something();
    throw err;
  }
});
```

**Comment by yusukebe** [comment, marked as answer]
It's implemented, resolved!
